### PR TITLE
Add SnapshotID field in AWSMachineClass

### PR DESCRIPTION
**Pull request by hardikdr**
**What this PR does / why we need it**: Enable support of snapshot-based volumes for machines in AWS.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Enable support of snapshot-based volumes for machines in AWS.
```

### CODE REVIEW

File: pkg/apis/machine/v1alpha1/types.go [hunk 1]
```diff
@@ -850,6 +850,18 @@ type AWSEbsBlockDeviceSpec struct {
...
 	Iops int64 `json:"iops,omitempty"`
 
+	// Identifier (key ID, key alias, ID ARN, or alias ARN) for a customer managed
+	// CMK under which the EBS volume is encrypted.
+	//
+	// This parameter is only supported on BlockDeviceMapping objects called by
+	// RunInstances (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html),
+	// RequestSpotFleet (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotFleet.html),
+	// and RequestSpotInstances (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotInstances.html).
+	KmsKeyID *string `json:"kmsKeyID,omitempty"`
```
Comment (prashanth26) [inline pkg/apis/machine/v1alpha1/types.go:860]: Why is this required? And if required, we will need some code changes also to accommodate the change right?

Comment (hardikdr) [reply pkg/apis/machine/v1alpha1/types.go:860]: This field is not required for the snapshot support. 
This will potentially be needed soon for the encryption of the worker-volumes. The unimplemented field doesn't seem to be hurting the flow, as the related implementation can always be added later.

Comment (prashanth26) [reply pkg/apis/machine/v1alpha1/types.go:860]: Okay.
